### PR TITLE
feat(ansible-doc): add completion

### PR DESCRIPTION
**Pull request by mamercad**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

New completion

**What is the current behavior? (You can also link to an open issue here)**

There is no completion for `ansible-doc` currently

**What is the new behavior (if this is a feature change)?**

There will be completion for `ansible-doc`

**Additional info:**

`man ansible-doc`

### CODE REVIEW

**Review by fedeci — CHANGES_REQUESTED** [review]

File: src/ansible-doc.ts [hunk 1]
```diff
@@ -0,0 +1,173 @@
+const allPluginsGenerator = (
+  suggestOptions?: Partial<Fig.Suggestion>
+): Fig.Generator => ({
```
[the remark this answers is not in the document]
Comment (mamercad) [reply src/ansible-doc.ts:3]: It enumerates the installed plugins like this:

image

File: src/ansible-doc.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+  name: "ansible-doc",
+  description: "Displays information on modules installed in Ansible libraries",
+  options: [
+    {
+      name: "--metadata-dump",
+      description:
+        "For internal testing only Dump json metadata for all plugins",
+      args: {
+        isOptional: true,
+        isDangerous: false,
```
[the remark this answers is not in the document]
Comment (mamercad) [reply src/ansible-doc.ts:24]: Cool; I'll push without them shortly.

File: src/ansible-doc.ts [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+): Fig.Generator => ({
+  script: "ansible-doc --list --json 2>/dev/null",
+  postProcess: function (output) {
+    const plugins = JSON.parse(output);
+    return Object.keys(plugins).map((key) => ({
+      name: key,
+      description: plugins[key],
+    }));
+  },
+});
```
Comment (fedeci) [inline src/ansible-doc.ts:12]: I mean, can we just refactor to this?
```suggestion
const allPluginsGenerator: Fig.Generator = {
  script: "ansible-doc --list --json 2>/dev/null",
  postProcess: function (output) {
    const plugins = JSON.parse(output);
    return Object.keys(plugins).map((key) => ({
      name: key,
      description: plugins[key],
    }));
  },
};
```

Comment (mamercad) [reply src/ansible-doc.ts]: I've committed this, but, it's failing `typecheck`. I don't know enough TypeScript to know what to do, can you help?

File: src/ansible-doc.ts [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+      args: {
+        isOptional: true,
+      },
+    },
+  ],
+  args: {
+    name: "plugin",
+    isDangerous: true,
+    isVariadic: true,
+    generators: allPluginsGenerator(),
```
Comment (fedeci) [inline src/ansible-doc.ts:154]: This was the cause of the typecheck failing @mamercad 
```suggestion
    generators: allPluginsGenerator,
```

Comment (mamercad) [reply src/ansible-doc.ts:154]: Ah; thanks, will push.

**Comment by mamercad** [comment]
Hrm; I rebased and the linter is exploding on all sorts of things that aren't mind 😄

**Review by fedeci — APPROVED** [review]
Thanks!

**Pull request merged**